DataStructureMemories/DataStructureMemories/DataStructureViewControllers/DetailViewController.swift

```diff
@@ -0,0 +1,137 @@
+//
+//  DetailViewController.swift
+//  DataStructureMemories
+//
+//  Created by Alina Zaitseva on 4/16/18.
+//  Copyright © 2018 Alina Zaitseva. All rights reserved.
+//
+import UIKit
+
+class DetailViewController: UIViewController {
+    
+    @IBOutlet weak var descriptionValue: UILabel!
+    @IBOutlet weak var gradientView: UIView!
+    @IBOutlet weak var halfHeightConstraint: NSLayoutConstraint!
+    @IBOutlet weak var fullHeightConstraint: NSLayoutConstraint!
+    @IBOutlet weak var heightConstraint: NSLayoutConstraint!
+    @IBOutlet weak var scrollView: UIScrollView!
+    
+    @IBOutlet weak var buttonToggle: UIButton!
+    @IBOutlet weak var buttonWiki: UIButton!
+    
+    public var srtuctureCell: DataStructMemory?
+    public var factoryEntity = ControlManagerFactory()
+    
+    var isShowingText = false
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        self.title = srtuctureCell?.titleOfString() ?? "There is no such title"
+        self.descriptionValue.text = srtuctureCell?.descrOfString() ?? "There is no such description"
+        gradientView.opacityGradient()
+    }
+    override func didReceiveMemoryWarning() {
+        super.didReceiveMemoryWarning()
+    }
+    
+    @IBAction func buttonShowText(_ sender: Any) {
+        let titleMore = "More"
+        let titleLess = "Less"
+        if isShowingText {
+            UIView.transition(with: scrollView, duration: 0.6, options: .transitionCrossDissolve, animations: {
+                self.buttonToggle.setTitle(titleMore, for: UIControlState.normal)
+                self.isShowingText = false
+                self.heightConstraint.priority = UILayoutPriority(rawValue: 999)
+                self.gradientView.isHidden = false
+            })
+        } else {
+            UIView.transition(with: scrollView, duration: 0.6, options: .transitionCrossDissolve, animations: {
+                self.buttonToggle.setTitle(titleLess, for: UIControlState.normal)
+                self.isShowingText = true
+                self.heightConstraint.priority = UILayoutPriority(rawValue: 250)
+                self.gradientView.isHidden = true
+            })
+        }
+        view.layoutIfNeeded()
+    }
+    
+    func giveWayToWKWebViewController() {
```

Comment:
You may define method as closure and pass it directly as UIAlertAction handler.

let giveWayToWKWebViewController: ((UIAlertAction) -> Void) { _ in
    //Your code goes here
}

let uiWebViewAction = UIAlertAction(title: "UIWebViewAction", style: .default, handler: giveWayToWKWebViewController)
actionSheetController.addAction(uiWebViewAction)